cinema/views.py

```diff
@@ -20,25 +20,52 @@
     OrderListSerializer,
 )
 
+from cinema.permissions import IsAdminOrIfAuthenticatedReadOnly
+
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet,
+):
     queryset = Genre.objects.all()
```

Comment:
You are using `all()` method directly on the model to define the queryset. This might lead to performance issues if the table contains a lot of records. Consider using a more specific queryset or a manager method.